lib/backend/k8s/resources/ippools_conversion.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2016 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resources
+
+import (
+	"encoding/json"
+	log "github.com/Sirupsen/logrus"
+	"strings"
+
+	"github.com/projectcalico/libcalico-go/lib/backend/k8s/thirdparty"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	kapi "k8s.io/client-go/pkg/api"
+)
+
+// ThirdPartyToIPPool takes a Kubrnetes ThirdPartyResource representation
+// of a Calico IP Pool and returns the equivalen IPPool object.
+func ThirdPartyToIPPool(t *thirdparty.IpPool) *model.KVPair {
+	v := model.IPPool{}
+	err := json.Unmarshal([]byte(t.Spec.Value), &v)
+	if err != nil {
+		panic(err)
```

Comment:
I've switched to a fatal log that gives more detail.  I think its still ok to shutdown here, since it's basically "someone wrote something wrong to the datastore, this should never happen".